library/core/src/ffi/c_str.rs

```diff
@@ -384,21 +385,37 @@ impl CStr {
     #[must_use]
     #[stable(feature = "cstr_from_bytes", since = "1.10.0")]
     #[rustc_const_stable(feature = "const_cstr_unchecked", since = "1.59.0")]
+    #[rustc_allow_const_fn_unstable(const_eval_select)]
     pub const unsafe fn from_bytes_with_nul_unchecked(bytes: &[u8]) -> &CStr {
-        // We're in a const fn, so this is the best we can do
-        debug_assert!(!bytes.is_empty() && bytes[bytes.len() - 1] == 0);
```

Comment:
I'd keep the debug assert here. It's useful even in a non-const context. Removing it here would regress what UB checks we do at runtime.

Reply:
Good point. I put it back with a short comment added inside the rt implementation since the const implementation performs a superset of this check already.